pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -193,6 +193,17 @@ func (a *index) constructService(svc *v1.Service, w *Waypoint) *workloadapi.Serv
 			Mode: workloadapi.LoadBalancing_STRICT,
 		}
 	}
+	ipFamily := workloadapi.IPFamilies_AUTOMATIC
```

Comment:
QQ - are we emulating the k8s `ipfamily` or `ipfamilypolicy` API here, or both?

It seems like mostly this is just emulating `ipfamilypolicy` semantics? Do we plan to hardcode the behavior, or expose `ipfamily/ipfamilypolicy` style APIs for the relevant resources we have, to match k8s?

"dualstack" is just two problems AFAICT:

- can you handle more than 1 IP assigned to (some resource with IPs).
- if you can, how do you order the list of IPs on that resource (k8s`ipfamily`)

The only bit we really care about as A Problem is the ordering of that IP list for non-K8S/Istio-only resources, right?

Reply:
ipFamilyPolicy. This is actually mostly about joining two resources (Service and Pod), and which IPs you should join. Since k8s already pre-filters the IPs (and assigns the IPs), its not like we need to do "which IPs do we assign" , etc

Reply:
Would we need to do that for stuff like SEs eventually, though?

Reply:
We really only _need_ this for headless services selecting pods, which there is no SE equivalent. So I don't think we really do